ext/pcre/php_pcre.c

```diff
@@ -2419,14 +2421,7 @@ PHP_FUNCTION(preg_replace_callback_array)
 		ZVAL_COPY_VALUE(&fci.function_name, replace);
 
 		replace_count += preg_replace_func_impl(&zv, str_idx_regex, /* regex_ht */ NULL, &fci, &fcc,
-			Z_STR_P(subject), Z_ARRVAL_P(subject),
-			limit, flags);
-
-		if (subject != return_value) {
```

Comment:
I don't get what this few lines were doing.. `return_value` should be `null` here, no?